### DISCUSSION TITLE
ELSO Brainstorm

### DISCUSSION THREAD

**Discussion by diasssavio** [nocontrolco/elso#4] [Ideas]
The discussion here is for everyone to freely comment feature ideas for the game that will (hopefully) be transformed into issues and (potentially) make into the game. This includes general game design and/or small improvements to already consolidated ideas.

**Comment by diasssavio** [comment]
The following ideas are more future ones, and have a significant impact on game future visual and core game physics itself. For the game physics, the impact and equations solving should not be exactly, though.

- As the ELSO (Electric Lightning Soccer Overdrive) name suggests, the whole game can be based on electromagnetism concepts. The arena would be electric charged as a way to influence the game dynamic.
- Use a dash/dodge mechanism inspired by [electromagnetic pulses](https://en.wikipedia.org/wiki/Electromagnetic_pulse). This way, vehicles could recharge their pulse when "attached" to a wall (arena or ball) and use it to "detach" from the wall.
- Use a boost/speed-up mechanism inspired by electricity, which is charged when walking down paths through the energized arena. An inspiring idea would be [MPD Thrusters](https://en.wikipedia.org/wiki/Magnetoplasmadynamic_thruster), which uses the force on a charged particle by an electromagnetic field to generate thrust. Once the vehicle is "detached" from a wall, it could also uses the thruster to generate lift.
- Have vehicles moving as [MagLev (Magnetic Levitation)](https://en.wikipedia.org/wiki/Maglev), levitating through the surfaces rather than actual wheels touching ground. They'd also have batteries to temporarily store energy to burst through and recharge by the arena.
- The ball would be made by a material that do not suffer from electrical or magnetic influence, taking forces only by gravity, Newton's second law, Euler's equation for rotational motion, & friction.

**Reply by herodrigues** [reply]
Based on those ideas, we can take inspiration from the game F-Zero regarding the vehicle levitation and movement.
- https://www.youtube.com/watch?v=BIe4AvKLSa4&t=1143s

**Reply by diasssavio** [reply]
Inspired on those ideas, the recharging paths may be fast or slow, depending on their location on the arena. And the recharging happens accordingly to the time spent on that path, instead of an instantly-discrete fashion.

**Comment by diasssavio** [comment]
How about a "[Viligante 8](https://vigilante8.fandom.com/pt-br/wiki/Vigilante_8:_2nd_Offense)"- like mode for the game as well?

Similar to the old never-continued PS1 game, we could have the cars battling with a set of weapons in different and wide maps. We could even add a lore for them. In Vigilante 8 2nd offense, there was a car from the future that may serve as inspiration for our cars as well.

 ![car from the future](https://github.com/nocontrolco/elso/assets/2767199/ba3b9c8e-d5a2-46a7-8b18-c5283cbafed0)

Each car would have the same mechanics as the football mode for moving, accelerating, dodging, etc. With the difference that they would also have a damage-meter that would make up for their life.

In fact, a damage-meter is a concept that eventually we may explore in the football game to trigger up demos, e.g.

Also, we could add external interactions in the maps to make the game more dynamic, as the original game had (giant monsters, teleports, boxes with weapons, water, machinery, life recharge...)
